.gitattributes

```diff
@@ -0,0 +1,4 @@
+* text=auto
+*.cs text=auto diff=csharp 
+*.sh text eol=lf
+*.sln text eol=crlf
```

Comment:
The actual diff in this PR is pretty useless because it's almost all line-endings other than this.

Reply:
Explicitly setting `eol=crlf` here means it will be CRLF on all platforms. Setting `auto` means that files will be checked in as LF but may be checked out as whatever the user/os prefers